### PR TITLE
DSL2 new release -- nf-core October hackathon sprint

### PR DESCRIPTION
**Pull request by yuukiiwa**
pre-game:
- [x] tested and slightly modified the original dsl2 implementation to [produce consistent results](https://github.com/nf-core/nanoseq/files/7441305/compare_dsl1_dsl2.tar.gz) with the dsl1 pipeline
- [x] solved bug on [alignment being skipped when providing `--skip_demultiplexing`](https://github.com/nf-core/nanoseq/issues/77#issuecomment-952106691)
    - Here is [the log of running test_bc_nodx.log](https://github.com/nf-core/nanoseq/files/7441282/bc_nodx_alignment_ran_through.log)

game plan:
- [x] replaced current local modules with `nf-core/modules`
- [x] updated `nanoseq` `dsl2` structure using `nf-core/rna-seq` as a reference
    - Here is [the spreadsheet of the local modules and subworkflows with links to their `versions.yml` update](https://docs.google.com/spreadsheets/d/1pPQkjkZ0d0Y5tM7j6NEV1kPCWl5GJM1p9Jg1km-jqNc/edit?usp=sharing)
- [x] tested the updated `nanoseq` with test data which [produced consistent results with the dsl1 pipeline](https://github.com/nf-core/nanoseq/files/7441320/compare_dsl1_newdsl2.tar.gz)
- [x] test the updated `nanoseq` with full data (`test_full`). The results are on `aws` machine, and downloading them will cost quite a bit, so it will be great if the result files can be transferred to the `awsmetadata` s3 bucket.
Screenshot from 2021-10-29 16-13-35

### CODE REVIEW

**Review by christopher-hakkaart — COMMENTED** [review]
Really fantastic! A huge effort and the changes look really good!

I found two very small things that (i think) are cosmetic rather than structural. The bones of the conversion look really great.

Because I don't think my comments are critical if you disagree or if I misunderstood we can resolve quickly and approve the merge.

File: modules/local/bam_rename.nf [hunk 1]
```diff
@@ -1,8 +1,5 @@
 // Import generic module functions
-include { initOptions; saveFiles; getSoftwareName } from './functions'
-
-params.options = [:]
-def options    = initOptions(params.options)
+include { saveFiles;; getProcessName } from './functions'
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply modules/local/bam_rename.nf]: Thanks for catching this! I have removed one of the `;`

File: nextflow_schema.json [hunk 2]
```diff
@@ -11,25 +11,22 @@
...
-                "protocol": {
-                    "type": "string",
-                    "fa_icon": "fas fa-vials",
-                    "description": "Specifies the type of data that was sequenced i.e. 'DNA', 'cDNA' or 'directRNA'."
+                    "format": "file-path",
+                    "mimetype": "text/csv",
+                    "pattern": "^\\S+\\.csv$",
+                    "schema": "assets/schema_input.json",
+                    "description": "Path to comma-separated file containing information about the samples in the experiment.",
+                    "help_text": "You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 3 columns, and a header row. See [usage docs](https://nf-co.re//usage#samplesheet-input).",
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply nextflow_schema.json]: I have updated the link to https://nf-co.re/nanoseq/1.1.0/usage

**Comment by yuukiiwa** [comment]
Thank you so much for reviewing this PR, @christopher-hakkaart! I have made the changes you suggested

**Review by christopher-hakkaart — APPROVED** [review]
Fantastic :)

**Pull request merged**